### PR TITLE
IWF-125: Limit error size return from API

### PR DESCRIPTION
**Pull request by lwolczynski**

### CODE REVIEW

File: service/interpreter/activityImpl.go [hunk 1]
```diff
@@ -180,16 +180,29 @@ func composeHttpError(
...
-		errMsg = errMsg[:maxL] + "..."
 
 		maxL = len(responseBody)
 		if maxL > 50 {
 			maxL = 50
+			responseBody = responseBody[:maxL] + "..."
 		}
-		responseBody = responseBody[:maxL] + "..."
 		errType = "1st-attempt-failure"
+	} else {
```
Comment (longquanzheng) [inline service/interpreter/activityImpl.go:192]: Nit: maybe nice to create a small func for the repeated code

File: service/interpreter/interfaces_mock.go [hunk 2]
```diff
@@ -0,0 +1,701 @@
...
+	reflect "reflect"
+	time "time"
+
+	gomock "github.com/golang/mock/gomock"
+	iwfidl "github.com/indeedeng/iwf/gen/iwfidl"
+	service "github.com/indeedeng/iwf/service"
+)
+
+// MockActivityProvider is a mock of ActivityProvider interface.
+type MockActivityProvider struct {
```
Comment (longquanzheng) [inline service/interpreter/interfaces_mock.go:18]: It’s better to name the file according so that it’s easier to find

Comment (lwolczynski) [reply service/interpreter/interfaces_mock.go:18]: Is there a better way of creating this mock? I did it this way:
```
mockgen -source=service/interpreter/interfaces.go -package=interpreter -destination=service/interpreter/interfaces_mock.go 
```

Comment (longquanzheng) [reply service/interpreter/interfaces_mock.go:18]: You can change the file name in the command or just rename the file in IDE. I am  not aware of other ways

Comment (lwolczynski) [reply service/interpreter/interfaces_mock.go:18]: I'm confused why change the name at all. I mocked the `interface` file, so I thought `interfaces_mock` is a good name

Comment (lwolczynski) [reply service/interpreter/interfaces_mock.go:18]: There are 13 mocked types in this file

Comment (longquanzheng) [reply service/interpreter/interfaces_mock.go:18]: Oh you are right! My bad. I thought the file only contains the mock for activity provider which you used in the tests.

File: service/interpreter/activityImpl_test.go [hunk 3]
```diff
@@ -88,3 +94,141 @@ func createCommands() ([]iwfidl.TimerCommand, []iwfidl.SignalCommand, []iwfidl.I
 	}
 	return validTimerCommands, validSignalCommands, internalCommands
 }
+
+func TestComposeHttpError_LocalActivity_LongErrorResponse(t *testing.T) {
```
Comment (stevo89519) [inline service/interpreter/activityImpl_test.go:98]: Some of these test initializations are using duplicated logic as well. It'd be nice if these were consolidated. Maybe a function that takes the repeat length and returns the httpResp?

Comment (lwolczynski) [reply service/interpreter/activityImpl_test.go:98]: I'll try to extract as much logic a to helper as I can

Comment (lwolczynski) [reply service/interpreter/activityImpl_test.go:98]: Refactored

File: service/interpreter/activityImpl.go (hunk 1 above)
Comment (lwolczynski) [reply service/interpreter/activityImpl.go:184]: Refactored

Comment (stevo89519) [reply service/interpreter/activityImpl.go:184]: Looks good

File: service/interpreter/activityImpl_test.go (hunk 3 above)
Comment (stevo89519) [reply service/interpreter/activityImpl_test.go:98]: Looks good

**Review by stevo89519 — APPROVED** [review]

**Review by longquanzheng — APPROVED** [review]

**Pull request merged**